erpnext/assets/doctype/asset/asset.py

```diff
@@ -681,7 +681,12 @@ def make_journal_entry(asset_name):
 
 	depreciation_cost_center, depreciation_series = frappe.db.get_value("Company", asset.company,
 		["depreciation_cost_center", "series_for_depreciation_entry"])
-	depreciation_cost_center = asset.cost_center or depreciation_cost_center
+
+	if asset.cost_center:
+		depreciation_cost_center = asset.cost_center
+
+	if asset.depreciation_expense_account:
+		depreciation_expense_account = asset.depreciation_expense_account
```

Comment:
Could you add comments to this function? It'll be a good way to enhance the quality of our code base